Source/Core/Core/HW/WiimoteEmu/Dynamics.cpp

```diff
@@ -398,7 +398,9 @@ Common::Quaternion GetRotationFromAcceleration(const Common::Vec3& accel)
 
 Common::Quaternion GetRotationFromGyroscope(const Common::Vec3& gyro)
 {
-  return Common::Quaternion{1, gyro.x / 2, gyro.y / 2, gyro.z / 2};
+  const auto length = gyro.Length();
+  return length ? Common::Quaternion::Rotate(length, gyro / length) :
+                  Common::Quaternion::Identity();
```

Comment:
IMO this should use `length != 0` instead of the implicit conversion. You could also theoretically use `Length2()` for that check and avoid a square root if the length is 0 and only compute the root if needed, but I suspect that a value of 0 is rare enough that doing that isn't worth the extra complexity.

Reply:
When the length-squared is non-zero we would still need the actual length here anyways so I don't think it's worth adding a `LengthSquared` function at the moment. I'll fix the implicit conversion.